### PR TITLE
Escape HTML from messages in error page

### LINKED ISSUE #4355: Cross Site Scripting（XSS）vulnerability in code-server
## OS/Web Information

- Web Browser: **firefox**
- Local OS: **Debian**
- Remote OS:**Debian**
- Remote Architecture:
- `code-server --version`: **v3.12.0**

## Steps to Reproduce

1.Open your browser and insert payload `/static/test%3Cmy_tag_efb4535077ba29aaca28167c491b4249/%3E%3Cimg%20src=x%3E%3Cscript%3Ealert(1)%3C/script%3E`

2.example: `http://127.0.0.1:8080/static/test%3Cmy_tag_efb4535077ba29aaca28167c491b4249/%3E%3Cimg%20src=x%3E%3Cscript%3Ealert(1)%3C/script%3E`

## Screenshot
image

## Notes

This issue can be reproduced in VS Code: Yes

### PR DESCRIPTION
**Pull request by mauri**
Fixes #4355

Escapes HTML contained in error messages before rendering the error page.

### CODE REVIEW

**Comment by jsjoeio** [comment]
WOW! Not only did you fix this, but you added tests! This has to be a first (at least in a while) 🎉

File: test/unit/node/routes/errors.test.ts
Comment (jsjoeio) [inline test/unit/node/routes/errors.test.ts:34]: Bummer we have to do this but I don't see any better alternatives

**Review by jsjoeio — APPROVED** [review]
Thanks so much for adding this and writing tests!

medal to luke

**Comment by jsjoeio** [comment]
The Docs preview is broken on forks at the moment but as long as everything else passes (it should), then we can merge!

**Comment by mauri** [comment]
Looks like E2E tests are failing to run, showing `Try re-installing playwright with "npm install playwright"`

**Comment by jsjoeio** [comment]
Do you mind rebasing onto `main` and pushing up? (I don't think maintainers can edit)

We pushed a fix for playwright so hopefully everything works 🤞

**Comment by mauri** [comment]
@jsjoeio rebased onto main branch 👍

**Comment by jsjoeio** [comment]
There's a high chance this doesn't work (e2e tests are failing on other PRs for some odd reason). We think [this](https://github.com/cdr/code-server/pull/4442) might fix but we'll see. So sorry for all the CI trouble!

**Comment by mauri** [comment]
@jsjoeio is there anything else I should do?

**Comment by jsjoeio** [comment]
@mauri thanks for the ping! Looks like @code-asher updated your branch. Once CI finishes, we should be good to merge!

**Comment by jsjoeio** [comment]
Not sure why macOS build is failing. Cancelling and re-running

image

**Pull request merged**
